### PR TITLE
Add retry with exponential backoff for monitor VM configuration

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

**Comment by persiaAziz** [comment]
> You are going to clean up (or squash) the commits, right?
Yes.. I will squash during merge

File: monitor/configureMonitorVM.sh [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
-#Updating apt-get
-apt-get -y update
-apt-get install -y python3-pip
-python -m pip install -U pip
-pip3 install pyhdb
+# disable exit on error
+set +e
+
+ExecuteCommand() {
+
```
Comment (tniek) [inline monitor/configureMonitorVM.sh]: nit: remove blank line

File: monitor/configureMonitorVM.sh [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+    retryCounter=1
+    while [ $retryCounter -le 3 ]
```
Comment (tniek) [inline monitor/configureMonitorVM.sh]: Given the flakiness we've observed, I'd increase this to 5.

File: monitor/configureMonitorVM.sh [hunk 3]
```diff
@@ -1,7 +1,32 @@
...
+    do
+        echo "Try no $retryCounter: Command: $1"
```
Comment (tniek) [inline monitor/configureMonitorVM.sh]: nit: replace "no" with "number" or "#"

File: monitor/configureMonitorVM.sh [hunk 4]
```diff
@@ -1,7 +1,32 @@
...
+        # Sleep to allow the servers to recover in case the server was unable to serve the request
+        sleep $retryCounter
```
Comment (tniek) [inline monitor/configureMonitorVM.sh]: Won't this just sleep $retryCounter seconds? Where is the exponential increase in delay here?

Comment (PakDLiu) [reply monitor/configureMonitorVM.sh]: ya, looks like linear backoff

File: monitor/configureMonitorVM.sh (hunk 3 above)
Comment (PakDLiu) [reply monitor/configureMonitorVM.sh]: +1

File: monitor/configureMonitorVM.sh (hunk 4 above)
Comment (persiaAziz) [reply monitor/configureMonitorVM.sh]: orry.. yes it is linear backoff. not exponential..

File: monitor/configureMonitorVM.sh (hunk 1 above)
Comment (persiaAziz) [reply monitor/configureMonitorVM.sh]: done

File: monitor/configureMonitorVM.sh (hunk 2 above)
Comment (persiaAziz) [reply monitor/configureMonitorVM.sh]: done

File: monitor/configureMonitorVM.sh (hunk 3 above)
Comment (persiaAziz) [reply monitor/configureMonitorVM.sh]: done

**Pull request merged**